azure-keyvault/porter.yaml

```
@@ -0,0 +1,157 @@
+name: hello-keyvault
+version: 0.1.2
+description: "Creates and azure keyvault using the az cli"
+registry: ghcr.io/asalbers
+dockerfile: Dockerfile.tmpl
+
+mixins:
+  - exec
+  - az
+
+install:
+  - az: 
+      description: "Logging into Azure..."  
+      arguments:
+        - login
+      suppress-output: true
+      flags:
+        debug: ""
+        service-principal:
+        username: "'{{bundle.credentials.AZURE_CLIENT_ID}}'"
+        password: "'{{bundle.credentials.AZURE_SP_PASSWORD}}'"
+        tenant: "'{{bundle.credentials.AZURE_TENANT_ID_OR_DNS}}'"
+  - az: 
+      description: "Setting the Azure subscription...."
+      suppress-output: true
+      arguments: 
+        - "account" 
+        - "set" 
+      flags:
+        subscription: "{{ bundle.credentials.AZURE_SUBSCRIPTION_ID}}"
+  - exec:
+      command: bash
+      description: "Creating or using the Azure resource group...."
+      arguments:
+        - ./upsertGroup.sh
+        - "{{bundle.parameters.AZURE_RESOURCE_GROUP_NAME}}"
+        - "{{bundle.parameters.AZURE_GROUP_REGION}}"
+  - exec:
+      command: bash
+      description: "Setting random string...."
+      arguments:
+        - ./helpers.sh
+        - random_string
+      outputs:
+        - name: random_string
+          regex: "(.*)"
+  - az:
+      description: "Creating the KeyVault..."
+      suppress-output: true
+      arguments:
+        - keyvault 
+        - create 
+      flags:
+        location: "eastus2"
```

Shouldn't this use the region parameter?

---

no, looking at the docs for the az cli it uses location as the region [docs](https://docs.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest#az_keyvault_create) Is that different from other az cli params?

---

What I mean is here the location flag is hard coded to eastus2, but the bundle has a parameter `AZURE_GROUP_REGION`. Shouldn't this location here match the same location used for the resource group? 

i.e. 
```suggestion
        location: "{{bundle.parameters.AZURE_GROUP_REGION}}"
```

---

oh, I can change that.